api/pacman-api-admin/src/main/java/com/tmobile/pacman/api/admin/repository/service/AbstractAccountServiceImpl.java

```diff
@@ -88,55 +97,95 @@ public AccountList getAllAccounts(String columnName, int page, int size, String
+        if (reqBody.getFilter() != null && !reqBody.getFilter().isEmpty()) {
+            StringBuilder clauseStr = new StringBuilder(" where ");
+            for (Map.Entry<String, Object> entry : reqBody.getFilter().entrySet()) {
+                if (entry.getKey() != null && entry.getValue() != null) {
+                    StringBuilder queryForAttribute = new StringBuilder("");
+                    if (entry.getKey().toString().equalsIgnoreCase("assets") || entry.getKey().toString().equalsIgnoreCase("violations")) {
+                        List<Map<String, Object>> rangeList = (List<Map<String, Object>>) entry.getValue();
+                        String queryWithDiffRangeOfEachAttribute = rangeList.stream().map(range -> "CAST(" + entry.getKey() + " AS decimal(10,2)) BETWEEN " + range.get("min") + " AND " + range.get("max"))
+                                .collect(Collectors.joining(" OR "));
+                        queryForAttribute = new StringBuilder("(").append(queryWithDiffRangeOfEachAttribute).append(")");
+                    } else {
+                        List<String> inpValues = (List<String>) entry.getValue();
+                        queryForAttribute = new StringBuilder("" + entry.getKey().toString() + " in (" + CommonUtils.joinListElement(inpValues) + ")");
+                    }
+                    accountQuery = accountQuery.append(clauseStr).append(queryForAttribute);
+                    clauseStr = new StringBuilder(" and ");
+                }
+            }
+        }
+        accountQuery = accountQuery.append(" order by " + sortElement + " " + sortOrder);
+        if(reqBody.getSize() > 0){
+            accountQuery = accountQuery.append(" limit " + reqBody.getPage() + ", " + reqBody.getSize());
+        }
+        return accountQuery == null ? "" : accountQuery.toString();
     }
 
     private AccountList convertToMap(Page<AccountDetails> entities) {
         AccountList accountList=new AccountList();
         List accountDetailsList= entities.getContent();
-
         Long elements=entities.getTotalElements();
         List<Map<String,String>> convertAccountDetails=new ArrayList<>();
         for(int i=0;i<accountDetailsList.size();i++){
-            Object[] ob= (Object[]) accountDetailsList.get(i);
             Map<String,String > accountMap=new HashMap<>();
-            accountMap.put("accountId",ob[0].toString());
-            accountMap.put("accountName",ob[1].toString());
-            accountMap.put("assets",ob[2].toString());
-            accountMap.put("violations",ob[3].toString());
-            accountMap.put("accountStatus",ob[4].toString());
-            accountMap.put("source",ob[5].toString());
-            accountMap.put("createdBy",ob[6]!=null?ob[6].toString():"");
-            accountMap.put("createdTime",ob[6]!=null?formatCreatedTime(ob[7]):"");
+            if(accountDetailsList.get(i) instanceof AccountDetails){
```

Comment:
This seems like duplicate code, consider extracting to a function

Reply:
convertToMap itself is a extracted function. It can be reused from other places.